File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/BOOTReceiver.java
package jp.superwooo.chaipon.lighttimeswitcher;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import java.net.Inet4Address;

public class BOOTReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        Log.d("LS", "BOOTReceiver triggered with action: " + action);
        if(intent.ACTION_BOOT_COMPLETED.equals(action)) {
            try {
                if(SchedulePreference.create(context).isEnabled())
                    AlarmScheduler.scheduleAll(context);
            } catch (SecurityException e) {
                Log.e("LS", "scheduling error because of security error");
            }
        }else{
            Log.d("LS", "BOOTReceiver ignored non-boot action: " + action);
        }
    }
}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/BOOTReceiver.kt
package jp.superwooo.chaipon.lighttimeswitcher

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import jp.superwooo.chaipon.lighttimeswitcher.AlarmScheduler.scheduleAll

class BOOTReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        Log.d("LS", "BOOTReceiver triggered with action: $action")
        if (Intent.ACTION_BOOT_COMPLETED == action) {
            try {
                if (SchedulePreference.Companion.create(context).isEnabled) scheduleAll(context)
            } catch (e: SecurityException) {
                Log.e("LS", "scheduling error because of security error")
            }
        } else {
            Log.d("LS", "BOOTReceiver ignored non-boot action: $action")
        }
    }
}
